drivers/QSPI.h

```diff
@@ -0,0 +1,262 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2015 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef MBED_QSPI_H
+#define MBED_QSPI_H
+
+#include "platform/platform.h"
+
+#if defined (DEVICE_QSPI) || defined(DOXYGEN_ONLY)
+
+#include "hal/qspi_api.h"
+#include "platform/PlatformMutex.h"
+#include "platform/SingletonPtr.h"
+#include "platform/NonCopyable.h"
+
+#define ONE_MHZ     1000000
+
+namespace mbed {
+    
+// Config/Mode Defines    
+/** QSPI Bus width Enum
+ */
+typedef enum qspi_config_bus_width {
+    QSPI_BUS_SINGLE,
+    QSPI_BUS_DUAL,
+    QSPI_BUS_QUAD,
+} qspi_config_bus_width_t;
+
+/** Address size Enum
+ */
+typedef enum qspi_config_address_size {
+    QSPI_ADDR_SIZE_NONE,
+    QSPI_ADDR_SIZE_8,
+    QSPI_ADDR_SIZE_16,
+    QSPI_ADDR_SIZE_24,
+    QSPI_ADDR_SIZE_32,
+} qspi_config_address_size_t;
+
+/** Alternative size Enum
+ */
+typedef enum qspi_config_alt_size {
+    QSPI_ALT_SIZE_NONE,
+    QSPI_ALT_SIZE_8,
+    QSPI_ALT_SIZE_16,
+    QSPI_ALT_SIZE_24,
+    QSPI_ALT_SIZE_32,
+} qspi_config_alt_size_t;
+
+/** QSPI Driver Return Status Enum
+ */
+typedef enum qspi_return_status {
+    QSPI_ERROR = -1, /**< Generic error >*/
+    QSPI_INVALID_PARAMETER = -2, /**< The parameter is invalid >*/
+    QSPI_SUCCESS = 0, /**< Function executed sucessfully  >*/
+} qspi_return_status_t;    
```

Comment:
After consideration, we should be using what HAL defines. `qspi_status_t` and other typedefs for arguments shall be used in the driver . This will remove all this redefinitions and mapping in the code file.